servicebroker/lib/routes/binding.js

```diff
@@ -20,57 +20,64 @@ module.exports = function(app) {
-                if (!error && (statusCode == 200 || statusCode == 201)) {
-                  t.commit();
-                  res.status(statusCode).json({});
+      models.sequelize.transaction().then(function(t) {
 
-                } else {
-                  logger.error("Fail to bind service when call api server, error=>" + error);
+        models.service_instance.findById(serviceInstanceId).then(function(result) {
+          if (result != null) {
+            models.binding.findAll({ where: { appId: appId } }).then(function(result) {
+              var length = result.length;
+              if (length === 0) { //no binding , so create one
+                models.binding.create({
+                  bindingId: bindingId,
+                  appId: appId,
+                  serviceInstanceId: serviceInstanceId,
+                  timestamp: new Date().getTime(),
+                  transaction: t
+                }).then(function(result) {
+                  apiServerUtil.attachPolicy(appId, policyJSON, function(error, response) {
+                    var statusCode = response.statusCode;
+                    if (!error && (statusCode == 200 || statusCode == 201)) {
+                      t.commit();
+                      res.status(statusCode).json({});
+
+                    } else {
+                      logger.error("Fail to bind service when call api server, error=>" + error);
+                      t.rollback();
+                      res.status(statusCode).json({});
+                    }
+                  });
+
+                }).catch(function(error1) { //catch findorcreate
+                  logger.error("Fail to bind service when add data to database, error=>" + error1);
                   t.rollback();
-                  res.status(statusCode).json({});
-                }
-              });
+                  if (error1 instanceof models.sequelize.UniqueConstraintError) {
+                    res.status(409).json({});
+                  } else {
+                    logger.error("Fail to handle request: " + JSON.stringify(req) + " with ERROR: " + error1);
+                    res.status(500).json({});
+                  }
 
-            }).catch(function(error1) { //catch findorcreate
-              logger.error("Fail to bind service when add data to database, error=>" + error1);
-              t.rollback();
-              if (error1 instanceof models.sequelize.UniqueConstraintError) {
-                res.status(409).json({});
-              } else {
-                logger.error("Fail to handle request: " + JSON.stringify(req) + " with ERROR: " + error1);
-                res.status(500).json({});
+                });
+              } else if (length > 1) { // an app has been bound to more than one service instance, this error should not exist
+                logger.error("");
+                res.status(499).json({ "description": messageUtil.getMessage("DUPLICATE_BIND", { "applicationId": appId }) });
```

Comment:
My point is that 400 is the right status code. If the CLI isn't displaying the message, we should open an issue against them to have it fixed.
CF_TRACE=true will always display it as a last resort.